diff -duNr openssl-1.0.1f-orig/crypto/ui/ui_openssl.c openssl-1.0.1f/crypto/ui/ui_openssl.c
--- openssl-1.0.1f-orig/crypto/ui/ui_openssl.c	2014-01-06 22:47:42.000000000 -0900
+++ openssl-1.0.1f/crypto/ui/ui_openssl.c	2014-03-27 12:27:13.000000000 -0900
@@ -190,9 +190,9 @@
 # undef  SGTTY
 #endif
 
-#if defined(linux) && !defined(TERMIO)
-# undef  TERMIOS
-# define TERMIO
+#if defined(linux) && !defined(TERMIOS)
+# undef  TERMIO
+# define TERMIOS
 # undef  SGTTY
 #endif
 
