--- texinfo-4.13/Makefile.in	2014-04-11 23:15:24.143459658 +0900
+++ texinfo-4.13-1/Makefile.in	2014-04-11 23:15:09.532267777 +0900
@@ -450,7 +450,7 @@
 @TOOLS_ONLY_FALSE@SUBDIRS = $(native_tools) gnulib/lib lib \
 @TOOLS_ONLY_FALSE@            install-info info makeinfo po util doc 
 
-@TOOLS_ONLY_TRUE@SUBDIRS = lib info install-info makeinfo util
+@TOOLS_ONLY_TRUE@SUBDIRS = gnulib/lib lib info install-info makeinfo util
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
